chapter2.adoc

```diff
@@ -44,41 +44,27 @@ When `mdbgen` is set to 0, the <<dbops, debug operations>> are disallowed and th
+
+[NOTE]
+When `dmprv`=1, the debugger memory access privilege matches the hart's privilege at Debug Mode entry. With `dmprv`, the debugger may modify the memory access privilege to VS/VU/U in Debug Mode when `mdbgen`=0 and `sdedbgalw`=1. However, the `dmprv` is still a debug feature, and it may be proposed to RISC-V Debug specification cite:[dbgspec] in the future to facilitate specific debug use cases. 
+
 [[dbgpriv]]
 [options="header"]
 [cols="15%,15%,35%,35%"]
 .External Debug Configuration and Privilege
 |============================================
-| mdbgen | sdedbgalw | Debug allowed privilege modes | Debug access privilege 
+| mdbgen | sdedbgalw | Debug allowed privilege modes | Debug access privilege (`dmprv`=0) 
 | 1      | Don't care      | All                      | M-mode
 | 0      | 1      | All except M                      | S-mode
 | 0      | 0      | None                              | N/A
 |============================================
 
-[[dbgaccpriv]]
-==== Debug Access Privilege
-
-The *debug access privilege* is defined as the privilege level granted to the external debugger when performing state accesses via the hart, such as Abstract Commands and Program Buffer execution.  Any attempt by the debugger to access state inaccessible to *debug access privilege* will return an error.
-
-[[prvvacc]]
-===== Configuring External Debugger Access Privileges
+==== Using EBREAK to Enter Debug Mode
 
-The `prv` and `v` fields in `dcsr` are updated with the current privilege mode on Debug Mode entry, and are used to set the new privilege mode on resume from Debug Mode.  The debugger may modify the `prv` and `v` fields, to alter the mode of execution upon resume, but the allowed values are constrained by the debugger privilege. 
-
-The maximum debug privilege level that can be configured in `prv` and `v` is shown in <<maxdbgpriv>>. On a write that attempts to write an illegal value to `prv` and/or `v`, the fields retain legal values. Illegal privilege levels include values higher than the maximum allowed debug privilege.
-
-[[maxdbgpriv]]
-[options="header"]
-.Maximum Allowed Resume Privilege Mode
-|=========================================
-| mdbgen | sdedbgalw | Maximum privilege allowed on resume
-| 1      | Don't care      | M
-| 0      | 1      | S(HS)
-| 0      | 0      | None
-|=========================================
-
-[NOTE]
-As the `prv` and `v` fields are Write Any Read Legal (WARL) fields, the external debugger is able to read back the written value to determine the maximum debug privilege level.  
+The EBREAK instruction can enter Debug Mode in the corresponding privilege level when the `ebreakm`, `ebreaks`, `ebreaku`, `ebreakvs`, and `ebreakvu` fields in `dcsr` are set to 1. However, this configuration will be ignored if the hart is running in a debug disallowed privilege level, in which case EBREAK will always raise breakpoint exception instead. 
```

Comment:
EBREAK works normal as specified in debug spec when external debug is allowed at running privilege level, when running in a debug disallowed privilege level, EBREAK will result in breakpoint exception
```suggestion
The EBREAK instruction can enter Debug Mode in the corresponding privilege level when the `ebreakm`, `ebreaks`, `ebreaku`, `ebreakvs`, and `ebreakvu` fields in `dcsr` are set to 1. However, this configuration will be ignored if the hart is running in a debug disallowed privilege level, in which case EBREAK will always raise breakpoint exception instead. 
```